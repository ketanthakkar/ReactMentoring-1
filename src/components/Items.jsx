import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Item from './Item';

const ItemsSection = styled.div`
    background-color: #fff;
`;

const ItemsWrapper = styled.div`
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    width: 900px;
    margin: 0 auto;
`;

const EmptyResults = styled.div`
    font-size: 50px;
    height: 400px;
    padding-top: 100px;
    width: 100%;
    text-align: center;
`;

class Items extends React.Component {
    render() {
        const {
            items,
            getItem,
            getItemsByGenre
        } = this.props;

        return (
            <ItemsSection>
                <ItemsWrapper>
                    {items && items.length ? (
                        items.map(
                            item => (
                                <Item
                                    key={item.id}
                                    item={item}
                                    getItem={getItem}
                                    getItemsByGenre={getItemsByGenre}
                                />
                            )
                        )
                    ) : (
                        <EmptyResults>No films found</EmptyResults>
                    )}
                </ItemsWrapper>
            </ItemsSection>
        )
    }
}

Items.propTypes = {
    items: PropTypes.array,
    getItem: PropTypes.func,
    getItemsByGenre: PropTypes.func,
};

Items.defaultProps = {
    items: [],
    getItem: null,
    getItemsByGenre: null,
};

export default Items;
