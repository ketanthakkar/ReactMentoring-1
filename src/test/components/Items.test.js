import React from 'react';
import { shallow } from 'enzyme';

import Items from 'components/Items';

describe('Items', () => {
    const items = shallow(<Items />);

    it('should be like snapshot', () => {
        expect(items).toMatchSnapshot();
    });
});
