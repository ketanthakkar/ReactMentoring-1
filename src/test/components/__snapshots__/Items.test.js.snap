// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should be like snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Items
    items={Array []}
    onItemClick={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Items__ItemsWrapper>
        <Items__EmptyResults>
          No films found
        </Items__EmptyResults>
      </Items__ItemsWrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Items__EmptyResults>
          No films found
        </Items__EmptyResults>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "No films found",
        },
        "ref": null,
        "rendered": "No films found",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Items__ItemsWrapper>
          <Items__EmptyResults>
            No films found
          </Items__EmptyResults>
        </Items__ItemsWrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Items__EmptyResults>
            No films found
          </Items__EmptyResults>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "No films found",
          },
          "ref": null,
          "rendered": "No films found",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
